lib/base_service.js

```diff
@@ -32,13 +33,38 @@ class BaseService {
         this._logger = null;
         this._metrics = null;
         this._ratelimiter = null;
-        this.serviceReturns = null;
+        this._serviceReturns = null;
+
+        // Avoid recursion if there are bugs in the logging code.
+        this._inLogger = false;
+
+        this._uncaughtExceptionHandler = (err) => {
+            this.logUnhandledException(err);
```

Comment:
Em, man... You've just removed this metod!

Reply:
Fixed

Reply:
Could be, but the fix hasn't been uploaded.

Usually, the way we work is that we first change/address all of the comments in code, then upload the new version and then comment with the outcomes for each specific comment. That way, it's easy to follow what is happening. In this instance, you say you fixed it, but you haven't uploaded the fix itself. At the same time, there are some other comments to be addressed throughout the code, so we don't know what is really happening.

Reply:
I was working on something else while I was waiting for feed back. I had not marked this resolved just added comment.

Reply:
And that is fine. My point was that it's better to do a pass on all of the comments that are clear/fixed once you uploaded the fixes/changes, as that makes it less messy and easier to follow. Otherwise, both the author and reviewers can eaily tell which comments have been addressed and which haven't.

Reply:
Just pushed the change and mark this resolved

Reply:
I saw you pushed the change, but that was beside the point of my comment. Please re-read this whole thread.